File: src/dependent/notifications/smtp_client.py
class SmtpClient:
    def __init__(self, host: str, port: int) -> None:
        if host is None or host == "":
            raise ValueError("SMTP host must be provided")
        if port <= 0:
            raise ValueError("SMTP port must be positive")
        self._host = host
        self._port = port

    def send(self, to: str, subject: str, body: str) -> str:
        if to is None or to == "":
            raise ValueError("Recipient must be provided")
        if subject is None:
            raise ValueError("Subject cannot be null")
        if body is None:
            raise ValueError("Body cannot be null")
        return "msg-" + str(abs(hash(to + subject + body + self._host + str(self._port))))

File: src/dependent/notifications/SmtpClient.java
package org.example.dependent.notifications;

public class SmtpClient {
    private final String host;
    private final int port;

    public SmtpClient(String host, int port) {
        if (host == null || host.isEmpty()) {
            throw new IllegalArgumentException("SMTP host must be provided");
        }
        if (port <= 0) {
            throw new IllegalArgumentException("SMTP port must be positive");
        }
        this.host = host;
        this.port = port;
    }

    public String send(String to, String subject, String body) {
        if (to == null || to.isEmpty()) {
            throw new IllegalArgumentException("Recipient must be provided");
        }
        if (subject == null) {
            throw new IllegalArgumentException("Subject cannot be null");
        }
        if (body == null) {
            throw new IllegalArgumentException("Body cannot be null");
        }
        return "msg-" + Math.abs((to + subject + body + host + port).hashCode());
    }
}


